core/object.h

```diff
@@ -484,10 +484,12 @@ class Object {
 	void _set_indexed_bind(const NodePath &p_name, const Variant &p_value);
 	Variant _get_indexed_bind(const NodePath &p_name) const;
 
-	void *_script_instance_bindings[MAX_SCRIPT_INSTANCE_BINDINGS];
-
 	void property_list_changed_notify();
 
+	friend class Reference;
+	uint8_t instance_binding_count;
```

Comment:
I think a bool may actually be better that you set to true, since it's never really decremented.  No need to use thread safety either if so.

Reply:
I don't see how a bool would remove the need for thread safety. BTW, is there a reason not to use a Vector for `_script_instance_bindings` instead of a raw array?

Reply:
ah nevermind, i misunderstood, yes, a bool won't do. just use an atomic operation for incrementing and it should be fine.